### PR TITLE
Fix/313 fix creating fault tree

### PR DESCRIPTION
**Pull request by kostobog**
@blcham 
@LaChope 
This PR is contains several issues:
#321 
#322 
#324
#354, #350
#313

### CODE REVIEW

File: src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx [hunk 1]
```diff
@@ -45,13 +46,12 @@ const FaultTreeAndSystemOverviewTable: FC<FaultTreeOverviewTableProps> = ({
   const { t } = useTranslation();
   const theme = useTheme();
 
+  const [, setSelectedSystem] = useSelectedSystem();
```
Comment (blcham) [inline src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx:49]: is this ok ?

File: src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx [hunk 2]
```diff
@@ -5,10 +5,11 @@ import { useTranslation } from "react-i18next";
 import useStyles from "./FaultTreeOverviewTable.styles";
 import { FaultTree } from "@models/faultTreeModel";
 import { useNavigate } from "react-router-dom";
-import { ROUTES, SELECTED_SYSTEM } from "@utils/constants";
+import {ROUTES} from "@utils/constants";
```
Comment (blcham) [inline src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx:8]: not going over prettier

File: src/hooks/useSelectedSystem.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
+import { createContext, useContext, useState } from "react";
+import {SELECTED_SYSTEM} from "@utils/constants";
+import { ChildrenProps } from "@utils/hookUtils";
+import {System} from "../models/systemModel";
```
Comment (blcham) [inline src/hooks/useSelectedSystem.tsx:5]: not used prettifier

File: src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx (hunk 2 above)
Comment (LaChope) [reply src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx:8]: I think there is an issue with Prettier in this repo... I have to trigger it manually (even if my settings show that it should be on every save + we are supposed to have a pre-commit hook that prevents these issues)

File: src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx (hunk 1 above)
Comment (LaChope) [reply src/components/dashboard/content/list/FaultTreeAndSystemOverviewTable.tsx:49]: yes

**Comment by blcham** [comment]
@kostobog for now max will work on this PR

**Comment by blcham** [comment]
@kostobog i guess we can close it **without merge** right (it was implemented elsewhere)

**Pull request closed without merging**